### PR TITLE
bugfix: Include semanticdb plugin path in java options processorpath

### PR DESCRIPTION
**Pull request by gersonsosa**
For projects that use javac "-processorpath" option append semanticdb plugin path to avoid compilation errors

Fixes:
#2059

### CODE REVIEW

**Comment by tgodzik** [comment]
Just got back to work this week, I am running the jobs and will take a look at the PR! Sorry for the delay.

File: frontend/src/test/resources/scala-java-processorpath/build.sbt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+bloopConfigDir in Global := baseDirectory.value / "bloop-config"
```
Comment (tgodzik) [inline frontend/src/test/resources/scala-java-processorpath/build.sbt:1]: Is this needed? It seems to be failing the tests and the defaults should work correctly

File: frontend/src/test/scala/bloop/bsp/BspBaseSuite.scala [hunk 2]
```diff
@@ -606,6 +606,31 @@ abstract class BspBaseSuite extends BaseSuite with BspClientTest {
     }
   }
 
+  def loadBspMetalsBuildFromResources(
+      buildName: String,
+      workspace: AbsolutePath,
+      logger: RecordingLogger,
+      bspClientName: String,
+      bloopExtraParams: BloopExtraBuildParams = BloopExtraBuildParams.empty
```
Comment (tgodzik) [inline frontend/src/test/scala/bloop/bsp/BspBaseSuite.scala:614]: You can just add it to the method below, should work fine

File: frontend/src/main/scala/bloop/data/Project.scala [hunk 3]
```diff
@@ -474,11 +474,33 @@ object Project {
       options ++ ysemanticdb ++ sourceRoot
     }
 
-    def enableJavaSemanticdbOptions(options: List[String]): List[String] = {
-      if (hasJavaSemanticDBEnabledInCompilerOptions(options)) options
+    def hasProcessorPath(options: List[String]): Boolean =
+      options.contains("-processorpath")
+
+    def includeSemnaticDBInProcessorPath(
```
Comment (tgodzik) [inline frontend/src/main/scala/bloop/data/Project.scala:480]:
```suggestion
    def includeSemanticdbInProcessorPath(
```
super minor

File: frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala [hunk 4]
```diff
@@ -477,6 +536,26 @@ class BspMetalsClientSpec(
...
+  }
+
+  private def semanticdbFileProject(
+      sourceFileName: String,
+      state: TestState,
+      projectName: String
+  ) = {
+    val project = state.build.getProjectFor(projectName).get
+    val classesDir = state.client.getUniqueClassesDirFor(project, forceGeneration = true)
+    val sourcePath = if (sourceFileName.startsWith("/")) sourceFileName else s"/$sourceFileName"
```
Comment (tgodzik) [inline frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala:555]: Why add `/` you already add it below?

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks for the great work! Added a couple of comments, but let's see if we can make the tests pass :1st_place_medal:

File: frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala (hunk 4 above)
Comment (gersonsosa) [reply frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala]: ah! I don't know, removed!

File: frontend/src/test/scala/bloop/bsp/BspBaseSuite.scala (hunk 2 above)
Comment (gersonsosa) [reply frontend/src/test/scala/bloop/bsp/BspBaseSuite.scala:614]: done!

File: frontend/src/main/scala/bloop/data/Project.scala [hunk 5]
```diff
@@ -474,11 +474,33 @@ object Project {
...
+        } else options
+      } else options
+
+    }
+
+    def enableJavaSemanticdbOptions(
+        options: List[String],
+        pluginPath: AbsolutePath
+    ): List[String] = {
+      val pluginOptions = includeSemnaticDBInProcessorPath(options, pluginPath)
```
Comment (tgodzik) [inline frontend/src/main/scala/bloop/data/Project.scala:497]:
```suggestion
      val pluginOptions = includeSemanticdbInProcessorPath(options, pluginPath)
```

Did you rename via Metals and it failed? :O

Comment (gersonsosa) [reply frontend/src/main/scala/bloop/data/Project.scala]: no I applied the suggestion directly in gh 😮‍💨 
sorry for the noise

**Comment by tgodzik** [comment]
I added you to triage role in the repo, so that I don't have to click approve every time. Let me know if you need help figuring out the tests!

**Comment by gersonsosa** [comment]
@tgodzik thank you, I think what's missing is adding bloop to the sample project 😅

**Comment by tgodzik** [comment]
I think we are almost there, but the windows tests are failing with:
```
--P:semanticdb:sourceroot:D:\a\bloop\bloop\frontend\src\test\resources\scala-java-processorpath
+-P:semanticdb:sourceroot:C:\Users\runneradmin\AppData\Local\Temp\bloop-test-workspace14905355409242014016
```

sourceroot is being set to the original resource directory and not the temporary workspace.

**Comment by tgodzik** [comment]
Weirdly enough it seems that on Windows we don't actually run within the created temporary workspace :thinking: 

I don't see any obvious reason for that, I can take a look on a windows machine tomorrow.

**Comment by gersonsosa** [comment]
@tgodzik I see, not running in a workspace is the reason for the difference in the test when asserting the workspace path.

I ran the tests on a windows machine, looks like the Compiler is unable to locate the semanticdb plugin on windows
I added some mode debugging and I get this from the compilation results
```
Compilation failed with results:
Failed(List(ProblemPerPhase(Problem(-1,Error,plug-in not found: semanticdb,,,Optional.empty),None)),None,10732,bloop.Compiler$$anon$3@776647f6)
```

After some trial and error the issue is that in windows you must use `;` instead of `:` to separate classpath or processorpath entries for javac.
I'll try to make changes to fix it later today.

**Comment by gersonsosa** [comment]
@tgodzik I'm not sure if the last failure of the pipeline is related to the changes, I would appreciate you to take a look and let me know how to proceed 👍🏼

**Review by tgodzik — APPROVED** [review]
LGTM! The failing test was not related.

**Pull request merged**